RCSwitch.cpp

```diff
@@ -436,6 +458,53 @@ char* RCSwitch::getCodeWordD(char sGroup, int nDevice, bool bStatus) {
 }
 
 /**
+* Encoding for Intertechno self-learning switch type
+* Protocol info found at https://www.sweetpi.de/blog/329/ein-ueberblick-ueber-433mhz-funksteckdosen-und-deren-protokolle
+* Tested with ITLS-16 remote control and ITLR-3500 socket adapter. The allSwitches flag seems to have no effect in this combo.
+*
+* @param remote  unique 26bit code of the remote control
+* @param allSwitches	apply command to all switches of the remote at once
+* @param nSwitchNumber  Number of switch button pair (1..16)
+* @param bStatus       Whether to switch on (true) or off (false)
+*
+* @return char[65], representing a binary code word of length 64
+*/
+char* RCSwitch::getCodeWordE(int remote, bool allSwitches, int nSwitchNumber, bool bStatus) {
+	static char sReturn[65];
+	int nReturnPos = 0;
+
+	for (int i = 25; i >= 0; i--) {
+		if (remote & (1 << i)) {
+			sReturn[nReturnPos++] = '1';
+			sReturn[nReturnPos++] = '0';
+		} else {
+			sReturn[nReturnPos++] = '0';
+			sReturn[nReturnPos++] = '1';
+		}
+	}
+	sReturn[nReturnPos++] = allSwitches ? '1' : '0';
+	sReturn[nReturnPos++] = allSwitches ? '0' : '1';
+	if (bStatus) {
+		sReturn[nReturnPos++] = '1';
+		sReturn[nReturnPos++] = '0';
+	} else {
+		sReturn[nReturnPos++] = '0';
+		sReturn[nReturnPos++] = '1';
+	}
+	for	(int i = 3; i >= 0; i--) {
+		if (nSwitchNumber & (1 << i)) {
+			sReturn[nReturnPos++] = '1';
+			sReturn[nReturnPos++] = '0';
+		} else {
+			sReturn[nReturnPos++] = '0';
+			sReturn[nReturnPos++] = '1';
+		}
+	}
+	sReturn[nReturnPos] = '\0';
+	return sReturn;
+}
```

Comment:
So all in all, this code makes it look as if for this remote, a single code bit is translated into four pulses (high-low-high-low) instead of two (high-low). 

With that interpretation, it becoes 32 bits of data. Hmmm.